src/interaction/FixedTripleListInteractionTemplate.hpp

```diff
@@ -145,16 +208,65 @@ inline real FixedTripleListInteractionTemplate<_AngularPotential>::computeEnergy
 
     const bc::BC &bc = *getSystemRef().bc;
     real e = 0.0;
-    for (FixedTripleList::TripleList::Iterator it(*fixedtripleList); it.isValid(); ++it)
+    real offs = getSystemRef().shearOffset;
+
+    if (offs != .0)
     {
-        const Particle &p1 = *it->first;
-        const Particle &p2 = *it->second;
-        const Particle &p3 = *it->third;
-        // const Potential &potential = getPotential(p1.type(), p2.type());
-        Real3D dist12 = bc.getMinimumImageVector(p1.position(), p2.position());
-        Real3D dist32 = bc.getMinimumImageVector(p3.position(), p2.position());
-        potential->computeColVarWeights(dist12, dist32, bc);
-        e += potential->_computeEnergy(dist12, dist32);
+        real Lx = bc.getBoxL()[0];
+        real Lz = bc.getBoxL()[2];
+        int xtmp;
+        real sqrlz_4 = Lz * Lz / 4.0, dpos;
+        for (FixedTripleList::TripleList::Iterator it(*fixedtripleList); it.isValid(); ++it)
+        {
+            const Particle &p1 = *it->first;
+            const Particle &p2 = *it->second;
+            const Particle &p3 = *it->third;
+            // const Potential &potential = getPotential(p1.type(), p2.type());
+            Real3D dist12, dist32;
+
+            Real3D dist_tmp(.0);
+            dpos = p1.position()[2] - p2.position()[2];
+            if (dpos * dpos > sqrlz_4)
+            {
+                dist_tmp[0] = (dpos > .0 ? -offs : offs);
+                xtmp = static_cast<int>(
+                    floor((p1.position()[0] + dist_tmp[0] - p2.position()[0]) / Lx + 0.5));
+                dist_tmp[0] -= (xtmp + .0) * Lx;
+                dist12 = bc.getMinimumImageVector(p1.position() + dist_tmp, p2.position());
+            }
+            else
+                dist12 = bc.getMinimumImageVector(p1.position(), p2.position());
```

Comment:
Code like this seems to be duplicated in a lot of places? Can they be unified?

Reply:
For angular forces, one needs to calculate two bond lengths. that's why you see duplicated code
ofc, we can have only one bc.getMinimumImageVector but the operation of +dist_tmp is then inevitable no matter whether the if-statement is satisfied.

Reply:
Just by looking into that one file I found 8 occurrences of more or less the same code. But yes, it is not your fault so let's skip it.

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L141-L152

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L154-L165

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L227-L238

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L240-L251

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L355-L366

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L368-L379

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L433-L444

https://github.com/espressopp/espressopp/blob/d6bf10050813529f59eb29d70b7a27d8c69c2915/src/interaction/FixedTripleListInteractionTemplate.hpp#L446-L457